onboarding_documentation/technical_documentation/metamist_docker/metamist_docker_tutorial.md

```diff
@@ -0,0 +1,783 @@
+
+The workflow management software we use at the CPG is called [Hail Batch](https://hail.is/docs/batch/service.html) - this allows us to specify a different Docker image for each step of a workflow, and metadata for each run records the name and version of each Image used. This is one of the ways in which CPG runs aim to be reproducible, e.g. if we need to go back and repeat a prior experiment we can be confident of the software used.
+
+[FastQE](https://github.com/fastqe/fastqe/) is a fun tool that mimics the output of FastQC (a more 'official' bioinformatic tool) but instead represents the quality scores with emojis. Clearly, FastQE is not intended for use in a production environment. However, it serves as an illustrative example of a tool currently missing from our library. To incorporate it, we need to construct a corresponding Docker image.
+
+Let's walk through an example of writing a Dockerfile using FastQE. You can refer to the [images](https://github.com/populationgenomics/images) repo to see how other images are built. **Note:** This is just an example for practice. When you're creating your own Dockerfile, you should try to do this yourself. However, to avoid duplication, please do not push these practice images to the repo.
+
+In this example, we'll use the `python:3.10-slim` image as a base image and install FastQE version 0.3.1 using `pip`.
+
+<details>
+  <summary>Click to see answer</summary>
+
+```Docker
+FROM python:3.10-slim
+
+ARG VERSION=${VERSION:-0.3.1}
+RUN pip install fastqe==${VERSION}
+```
+
+</details>
+<br>
+
+Save the above in a file named `Dockerfile` and push it to the `images` repo, create a PR and as is standard practice in CPG you will have to nominate someone to review this PR so that it can be merged into main and the image can be used by `analysis-runner` when we run the job.
+
+CLI command to build the image:
+
+```bash
+docker build -t fastqe_image:1.0.0 . --platform=linux/amd64
+```
+
+Note that we need to specify the platform as linux/amd64 because we are building the image on a Mac. If you are building the image on a Linux machine, you can omit the --platform flag. However, when creating an image for our infrastructure, this won't be necessary. Docker files should be platform agnostic, meaning that if a build is successful on both your local machine and the GitHub server, the resulting images should be identical. This is beneficial for testing purposes, but it's not a prerequisite for creating images within our infrastructure.
+
+
+
+## 3. Write a script to run FastQE
+
+In this section, we'll craft a script that leverages Hail Batch and the FastQE image we just made. This script will execute the FastQE tool on the `.fastq` files that we previously retrieved from Metamist. After completing this section you will have an understanding of how Hail Batch uses Docker images to run analyses and how to write a script that uses Hail Batch to run an analysis.
+
+The basic outline of our script is going to be:
```

Comment:
Are we intending for users to run this, if so we should add instructions on branching, and committing changes so they can follow along.